pkgs/top-level/all-packages.nix

```diff
@@ -16310,17 +16310,25 @@ in
   linuxManualConfig = makeOverridable (callPackage ../os-specific/linux/kernel/manual-config.nix {});
 
   # Derive one of the default .config files
-  linuxConfig = { src, makeTarget ? "defconfig", name ? "kernel.config" }:
-    stdenv.mkDerivation {
-      inherit name src;
-      buildPhase = ''
-        set -x
-        make ${makeTarget}
-      '';
-      installPhase = ''
-        cp .config $out
-      '';
-    };
+  linuxConfig = let
+    kernelArch = stdenv.hostPlatform.kernelArch;
+  in buildPackages.callPackage ({
```

Comment:
This looks wrong I think. We don't want to use buildPackage.callPackage, because you will have no access to buildInputs (even though none are needed here).

Reply:
/cc @Ericson2314

Reply:
Yes this is wrong. My guess is you want `stdenvNoCC` + `depsBuildBuild = [ buildPackages.stdenv.cc ];` + `HOST_CC = ..` See how linux-headers does it.